docs/source-app/core_api/lightning_app/dynamic_work_content.rst

```diff
@@ -0,0 +1,199 @@
+***************************************
+What Dynamic LightningWork does for you
+***************************************
+
+Dynamic LightningWork (Work) changes the resources your application uses while the application is running (aka at runtime).
+
+For example, imagine you want to create a research notebook app for your team. You want every member to be able to create multiple `JupyterLab <https://jupyter.org/>`_ sessions on their hardware of choice.
+
+To allow every notebook to choose hardware, it needs to be set up in it's own :class:`~lightning_app.core.work.LightningWork`, but you can't know the number of notebooks user will need in advance. In this case you'll need to add ``LightningWorks`` dynamically at run time.
+
+----
+
+*****************
+Use Dynamic Works
+*****************
+
+Dynamic Works should be used anytime you want change the resources your application is using while it is running (aka at runtime).
+
+You're usually going to use the ``start`` and ``stop`` methods together.
+
+----
+
+Add a Dynamic Work
+^^^^^^^^^^^^^^^^^^
+
+There are a couple of ways you can add a dynamic Work:
+
+- Option 1: Attach your components in the **run** method using the Python functions.
+- Option 2: Use the Lightning built-in classes :class:`~lightning.structures.Dict` or :class:`~lightning.structures.List`.
+
+.. note:: Using the Lightning built-in classes is usually easier to read.
+
+----
+
+**OPTION 1:** Attach your components in the run method of a flow using the Python functions **hasattr**, **setattr**, and **getattr**:
+
+.. code-block:: python
+
+    class RootFlow(lapp.LightningFlow):
+
+        def run(self):
+
+            if not hasattr(self, "work"):
+		# The `Work` component is created and attached here.
+                setattr(self, "work", Work())
+	    # Run the `Work` component.
+            getattr(self,  "work").run()
+
+**OPTION 2:** Use the built-in Lightning classes :class:`~lightning_app.structures.Dict` or :class:`~lightning_app.structures.List`
+
+.. code-block:: python
+
+    from lightning_app.structures import Dict
+
+    class RootFlow(lapp.LightningFlow):
+
+        def __init__(self):
+            super().__init__()
+            self.dict = Dict()
+
+        def run(self):
+            if "work" not in self.dict:
+			# The `Work` component is attached here.
+                self.dict["work"] = Work()
+            self.dict["work"].run()
+
+----
+
+Stop a Work
+^^^^^^^^^^^
+Stop a work when you are concerned about cost.
+
+To stop a work, use the work ``stop`` method:
+
+.. code-block:: python
+
+    class RootFlow(L.LightningFlow):
+
+        def __init__(self):
+            super().__init__()
+            self.work = Work()
+
+        def run(self):
+            self.work.stop()
```

Comment:
can we call it from within the work? using `self.stop`?

Reply:
This is meant to be used from the flow. We need to add a check to avoid this actually.